it/pages/agent/ConfirmEmailPageSpec.scala

```diff
@@ -0,0 +1,182 @@
+
+          "Render the Confirm Email view" in {
+
+            def show(): WSResponse = get(path, formatNotificationsEmail(None) ++ formatReturnUrl)
+
+            given.agent.isSignedUpToAgentServices
+
+            When("I call the Confirm email page with no notification email in session")
+
+            val res = show()
+
+            res should have(
+              httpStatus(SEE_OTHER),
+              redirectURI(controllers.agent.routes.CapturePreferenceController.show().url)
+            )
+          }
+        }
+
+        "there is a notification email in session" should {
+
+          "Render the Confirm Email view" in {
+
+            def show(): WSResponse = get(path, formatNotificationsEmail(Some(notificationsEmail)) ++ formatReturnUrl)
+
+            given.agent.isSignedUpToAgentServices
+
+            When("I call the Confirm email page with a notification email in session")
+            val res = show()
+
+            res should have(
+              httpStatus(OK),
+              elementText("h1")("Confirm the new email address")
+            )
+          }
+        }
+      }
+    }
+  }
+
+  "Calling the .isEmailVerified action" when {
+
+    "the user is an Agent" when {
+
+      "the Agent is signed up for HMRC-AS-AGENT" when {
+
+        "there is a verified notification email in session" should {
+
+          "Render the Select client VRN view" in {
+
+            val path = "/is-email-verified"
+
+            def isEmailVerified(): WSResponse = get(path, formatNotificationsEmail(Some(notificationsEmail)) ++
+              formatReturnUrl)
+
+            given.agent.isSignedUpToAgentServices
+
+            And("I stub an email verified successful response from EmailVerificationService")
+            EmailVerificationStub.stubEmailVerified
+
+            When("I call the isEmailVerified controller action with a verified notification email in session")
+
+            val res = isEmailVerified()
+
+            res should have(
+              httpStatus(SEE_OTHER),
+              redirectURI(controllers.agent.routes.SelectClientVrnController.show().url)
+            )
+          }
+        }
+
+        "there is a non-verified notification email in session" should {
+
+          "Render the Verify Email view" in {
+
+            val path = "/is-email-verified"
+
+            def isEmailVerified(): WSResponse = get(path, formatNotificationsEmail(Some(notificationsEmail)) ++
+              formatReturnUrl)
+
+            given.agent.isSignedUpToAgentServices
+
+            And("I stub an email not verified response from EmailVerificationService")
+            EmailVerificationStub.stubEmailNotVerified
+
+            When("I call the isEmailVerified controller action with a non-verified notification email in session")
+
+            val res = isEmailVerified()
+
+            res should have(
+              httpStatus(SEE_OTHER),
+              redirectURI(controllers.agent.routes.VerifyEmailController.sendVerification().url)
```

Comment:
What do you think about having a title / header check here as well? Just to make sure that the correct view is loaded?